components/membership/membership-http-rpc-impl/src/main/kotlin/net/corda/membership/impl/httprpc/v1/MGMRestResourceImpl.kt

```diff
@@ -436,6 +477,76 @@ class MGMRestResourceImpl internal constructor(
             ruleId: String
         ) = deleteGroupApprovalRule(holdingIdentityShortHash, ruleId, PREAUTH)
 
+        override fun viewRegistrationRequests(
+            holdingIdentityShortHash: String,
+            requestSubjectX500Name: String?,
+            viewHistoric: Boolean,
+        ): Collection<RestRegistrationRequestStatus> {
+            val requestSubject = requestSubjectX500Name?.let {
+                MemberX500Name.parse("requestSubjectX500Name", it)
+            }
+            return handleCommonErrors(holdingIdentityShortHash) {
+                mgmOpsClient.viewRegistrationRequests(
+                    ShortHash.parseOrThrow(holdingIdentityShortHash),
```

Comment:
```suggestion
                    it,
```

`handleCommonErrors` parses `holdingIdentityShortHash` and passes it in to the lambda so you don't have to parse it again.